components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/NotificationSender.java

```diff
@@ -18,16 +18,45 @@
 
 package org.wso2.carbon.identity.mgt;
 
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
 
 /**
  * email sender this creates a new task in thread pool for each email sending request
  */
 public class NotificationSender {
 
-    private static ExecutorService threadPool = Executors.newFixedThreadPool(5);
+    private static ExecutorService threadPool = null;
     NotificationSendingModule module;
+    private static final Log log = LogFactory.getLog(NotificationSender.class);
+
+    static {
+
+        IdentityMgtConfig identityMgtConfig = IdentityMgtConfig.getInstance();
+        if (identityMgtConfig != null) {
+
+            int threadPoolSize = identityMgtConfig.getNotificationSendingThreadPoolSize();
+            int notificationSendingTimeout = identityMgtConfig.getNotificationSendingTimeout();
+
+            if (threadPoolSize <= 0) {
+                threadPoolSize = 5;
+            }
+
+            Executors.newFixedThreadPool(threadPoolSize);
+            
+            if (notificationSendingTimeout > 0) {
+                try {
+                    threadPool.awaitTermination(notificationSendingTimeout * 1000, TimeUnit.MILLISECONDS);
```

Comment:
Will cause a NPE since threadPool initialized to null and haven't assigned any other value.